application/src/main/java/org/togetherjava/tjbot/commands/moderation/WhoIsCommand.java

```diff
@@ -0,0 +1,136 @@
+package org.togetherjava.tjbot.commands.moderation;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.Role;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.components.Button;
+import net.dv8tion.jda.api.interactions.components.ButtonStyle;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class WhoIsCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(WhoIsCommand.class);
+
+    private static final String USER_OPTION = "user";
+    private static final String HIDE_MEMBER_INFO = "hide_member_info";
+    private static final DateTimeFormatter DATE_TIME_FORMAT =
+            DateTimeFormatter.ofPattern("E, MMMM d, u, HH:mm:ss");
+
+    public WhoIsCommand() {
+        super("whois", "who is the given user?", SlashCommandVisibility.GUILD);
+
+        getData().addOption(OptionType.USER, USER_OPTION, "the user to look up", true);
+        getData().addOption(OptionType.BOOLEAN, HIDE_MEMBER_INFO,
+                "whenever to show the member info of this user", false);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull final SlashCommandEvent event) {
+        OptionMapping userOption = Objects.requireNonNull(event.getOption(USER_OPTION),
+                "The given user option cannot be null");
+
+        User user = userOption.getAsUser();
+        Member member = userOption.getAsMember();
+
+        if (null != member) {
+            OptionMapping hideMemberInfoOption = event.getOption(HIDE_MEMBER_INFO);
+
+            if (null != hideMemberInfoOption && hideMemberInfoOption.getAsBoolean()) {
+                handleWhoIsUser(event, user);
+            } else {
+                handleWhoIsMember(event, member, user);
```

Comment:
These method overloads are kinda confusing and definitely not obvious by just reading the signature. I.e., it is not obvious that one shows member extra info and the other not, just by passing in different arguments.

I think it is worth renaming them and only passing the relevant parameter (user can be retrieved from member anyways).

Also, both methods basically build an embed and the sending section is duplicated. It might be better to change them from standalone-handle methods to methods that just compute and return the embed and then doing the sending bit here again.

Maybe we can also append the member specific details (based on the flag) at the end of the embed build by the version that hides them.

So in a nutshell we would aim for sth like
```java
EmbedBuilder embed;
if (isUserMember) {
  embed = createResponseNotMember(user);
} else {
  embed = createResponseMember(member);
  if (isShowDetails) {
    appendDetails(embed);
  }
}

event.replyEmbeds(embed.build()).queue();

Reply:
Thing I struggled with was readability, I'd want join date and registration date to be after each other.
But other info might come after that too, so keeping it in a sensible order while also splitting things up into 2 methods, doesn't seem possible.

Take a look at the examples, `is bot` and `is booster` are after each other too as example ^^